spring-boot-project/spring-boot-tools/spring-boot-configuration-processor/src/json-shade/java/org/springframework/boot/configurationprocessor/json/JSONTokener.java

```diff
@@ -203,8 +203,7 @@ public String nextString(char quote) throws JSONException {
 			int c = this.in.charAt(this.pos++);
 			if (c == quote) {
 				if (builder == null) {
-					// a new string avoids leaking memory
-					return new String(this.in.substring(start, this.pos - 1));
+					return this.in.substring(start, this.pos - 1);
```

Comment:
I'm a bit confused about "a new string avoids leaking memory" - not sure how this is possible?